src/util.ts

```diff
@@ -0,0 +1,31 @@
+export const b64Decode = (function(): (s: string) => string {
+  if (isNodejs()) {
+    return s => Buffer.from(s, 'base64').toString('binary');
+  }
+  return window.atob;
+})();
+
+export const generateID: () => string = (() => {
+  if (isNodejs()) {
+    const crypto = require('crypto');
+    return () => {
+      return crypto.randomBytes(16).toString('base64');
+    };
+  } else {
+    return () => {
+      const arr = new Uint8Array(16);
+      crypto.getRandomValues(arr);
+      //    note this only works for short sequences because it is recursive
+      return btoa(String.fromCharCode.apply(null, Array.from(arr.values())));
+    };
+  }
+})();
```

Comment:
Making a note-to-self that we should keep an eye on the growth of docs with unique session IDs. Probably not a big deal if we GC.

Reply:
Ah yes didn't consider that, GC will indeed help tho